src/main/java/org/schemaspy/output/html/mustache/diagrams/MustacheOrphanDiagramFactory.java

```diff
@@ -77,7 +79,7 @@ public List<MustacheTableDiagram> generateOrphanDiagrams(List<Table> orphanTable
             File dotFile = orphanDir.resolve(dotBaseFilespec + ".1degree.dot").toFile();
 
             try (PrintWriter dotOut = Writers.newPrintWriter(dotFile)) {
-                dotProducer.writeOrphan(table, dotOut);
+                new DotOrphanFormatter(dotConfig).writeOrphan(table, dotOut);
```

Comment:
Yes, I also want to move in that direction. But I think we should do it in small steps.

> The dot-file generator can no longer be injected.

There was no injection to begin with. The constructor parameter was a dependency on a concrete implementation. (Not just a concrete class, but also very specific details.)

> This feels wrong.

I'm guessing that is because of `new`. It was already there, just obscured by a class that didn't really do anything.

Reply:
DotFormatter is not final.

But extracting an interface would've been a long way around? Since it's injected it could just have had a different signature like write without the Orphan and it could be left alone. While the implementation would evolv to a single implementation for any table?

Reply:
That is correct, but that doesn't make it a "good" dependency in terms of abstraction.

An interface would have been nice, but I decided it wasn't worth the trade-off at this time. The method parameters are too difficult mock/substitute, and I don't want to add more indirection. (See also the comment below.)

Reply:
First rule of programming all code is bad.

Reply:
"All code is bad, but some code is worse than other?" 🙂

Reply:
Just you wait and see, I'm cooking a a sh*t storm